CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+
+Sometimes you might need to change upstream dependent package(s) to be able to submit a meaningful change. We are using [`staged.dependencies`](https://github.com/openpharma/staged.dependencies) functionality to simulate a `monorepo` behavior. The configuration is already in the [`staged_dependencies.yaml`](staged_dependencies.yaml) file. You just need to name the feature branches identically. This is the only exception from branch naming convention described above. Please read the package manual for more details.
+
+### Recommended development environment & tools
+
+#### R & package versions
+
+We continuously test our packages against the newest R version as well as a given package dependencies. We recommend to set-up your working environment in the same way. You can find all the details in any of the actions.
+
+If you find out any bugs on the older version of dependencies - please create appropriate bug ticket.
+
+#### `pre-commit`
+
+We highly recommend the [`pre-commit`](https://pre-commit.com/) tool combined with [`R hooks for pre-commit`](https://github.com/lorenzwalthert/precommit) to execute some of the checks prior committing and pushing. The configuration is already there in a repository ([here](.pre-commit-config.yaml)). Please, follow the installation guide on the official [`pre-commit` page](https://github.com/lorenzwalthert/precommit) and the [`GitHub` `readme` page](https://github.com/lorenzwalthert/precommit#installation) for the R hooks.
+
+## Style guide
+
+This repository follows standard [`tidyverse` style guide](https://style.tidyverse.org/) and it's being checked against it by [`lintr`](https://github.com/r-lib/lintr). There are some slight modifications to its default settings available in the [`.lintr`](.lintr) file.
+
+Although it allows for some flexibility - we recommend sticking to the style of the existing code.
+
+Please note that there is a style and also `lintr` check in place that will validate your code.
+
+## Recognition model
+
+Any contribution is highly welcomed and appreciated. While all the contribution data is already there in GitHub repository insights feature, we introduced some objective rules to recognize a _significant_ contribution so as to became a package author:
+
+- Minimum 5% of lines of code authored (determined by `git blame` query) OR
+- Being at the top 5 contributors in terms of number of commits OR lines added OR lines removed.
+
+The package maintainer also reserve the rights to manually adjust the criteria to recognize contributions.
+
+## Any questions
+
+If you have any further questions regarding contribution - please reach out to the repository maintainer!
+
+## Note
+
+This is a template applied across multiple repositories within our organization. You might find additional, repository specific rules in the [`CONTRIBUTING_ADDITIONAL.md`](CONTRIBUTING_ADDITIONAL.md) file. If it's missing then it means that there is no additional considerations to the above.
```

Comment:
Does it make sense for you guys? Do you think we need it?
To me, there are some repo-specific rules e.g. in `tern` (specific naming convention, templating, etc.). I just want to open a door to specifying this while keeping the base identical across the repos. Will you use it or it's too much? @shajoezhu @gogonzo